src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -391,6 +401,75 @@ protected virtual void Initialize()
             }
         }
 
+        // Scan the extensions directory and Load custom extension.
+        // $$$ - need to finalize this policy. Currently enabled via an AppSettting. 
+        private void LoadCustomExtensions(JObject hostConfig)
+        {
+            var bindingRoot = ConfigurationManager.AppSettings["BYOB_Path"];
+            if (!string.IsNullOrWhiteSpace(bindingRoot))
+            {
+
+                var config = this.ScriptConfig.HostConfig;
+
+                foreach (var dir in Directory.EnumerateDirectories(bindingRoot))
+                {
+                    foreach (var path in Directory.EnumerateFiles(dir, "*.dll"))
+                    {
+                        // How to find the extension assembly? Don't want to load every one to inspect it. 
+                        // $$$ manifest file? scan for any assembly with 
+                        if (!path.ToLowerInvariant().Contains("extension"))
```

Comment:
I understand this logic is likely to change. But for now, we probably want to inspect the file name (I'm assuming that was your intention) as this would always be true if the root folder contains the string `extension` (which will most likely be the case)